cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -106,7 +122,7 @@ public AppFabricServer(CConfiguration cConf, SConfiguration sConf,
                          ProvisioningService provisioningService,
                          BootstrapService bootstrapService,
                          SystemAppManagementService systemAppManagementService,
-                         TransactionRunner transactionRunner) {
+                         TransactionRunner transactionRunner, MetadataStorage metadataStorage) {
```

Comment:
we don't want to write directly to a store from this AppFabricServer. This should mostly be a thin wrapper around other services, so that there is a clean separation of responsibility and logic. This looks more like something that should run in the metadata system service instead of app fabric.